tesseract-server/src/handlers/logic_layer/geoservice.rs

```diff
@@ -0,0 +1,82 @@
+use std::str;
+
+use actix_web::{
+    client,
+    HttpMessage,
+};
+use failure::{Error, format_err};
+use futures::future::Future;
+use serde_derive::Deserialize;
+
+
+#[derive(Debug, Clone, PartialEq, Deserialize)]
+pub struct GeoServiceResponseJson {
+    pub geoid: String,
+    pub level: String,
+}
+
+
+pub enum GeoserviceQuery {
+    Neighbors,
+    Children,
+    Parents,
+    Intersects,
+    Distance,
+}
+
+
+/// Queries geoservice for geo cuts resolution.
+pub fn query_geoservice(
+    geoservice_url: &str,
+    geoservice_query: &GeoserviceQuery,
+    geo_id: &str
+) -> Result<Vec<GeoServiceResponseJson>, Error> {
+    let mut query_url: String;
+
+    if &geoservice_url[geoservice_url.len()-1..] == "/" {
```

Comment:
you can do `let query_url =` and then the whole match block, instead of initializing then assigning.

Reply:
Also, I think that you should use the Url crate (which is probably reexported through actix) in order to handle the url robustly. Let me know what you think.